install.ps1

```diff
@@ -486,43 +498,55 @@ function Install-Scoop {
     # Enable TLS 1.2
     Optimize-SecurityProtocol
 
-    # Download scoop zip from GitHub
-    Write-InstallInfo "Downloading..."
+    # Download scoop from GitHub
+    Write-InstallInfo "Downloading ..."
     $downloader = Get-Downloader
-    # 1. download scoop
-    $scoopZipfile = "$SCOOP_APP_DIR\scoop.zip"
-    if (!(Test-Path $SCOOP_APP_DIR)) {
-        New-Item -Type Directory $SCOOP_APP_DIR | Out-Null
-    }
-    Write-Verbose "Downloading $SCOOP_PACKAGE_REPO to $scoopZipfile"
-    $downloader.downloadFile($SCOOP_PACKAGE_REPO, $scoopZipfile)
-    # 2. download scoop main bucket
-    $scoopMainZipfile = "$SCOOP_MAIN_BUCKET_DIR\scoop-main.zip"
-    if (!(Test-Path $SCOOP_MAIN_BUCKET_DIR)) {
-        New-Item -Type Directory $SCOOP_MAIN_BUCKET_DIR | Out-Null
-    }
-    Write-Verbose "Downloading $SCOOP_MAIN_BUCKET_REPO to $scoopMainZipfile"
-    $downloader.downloadFile($SCOOP_MAIN_BUCKET_REPO, $scoopMainZipfile)
-
-    # Extract files from downloaded zip
-    Write-InstallInfo "Extracting..."
-    # 1. extract scoop
-    $scoopUnzipTempDir = "$SCOOP_APP_DIR\_tmp"
-    Write-Verbose "Extracting $scoopZipfile to $scoopUnzipTempDir"
-    Expand-ZipArchive $scoopZipfile $scoopUnzipTempDir
-    Copy-Item "$scoopUnzipTempDir\scoop-*\*" $SCOOP_APP_DIR -Recurse -Force
-    # 2. extract scoop main bucket
-    $scoopMainUnzipTempDir = "$SCOOP_MAIN_BUCKET_DIR\_tmp"
-    Write-Verbose "Extracting $scoopMainZipfile to $scoopMainUnzipTempDir"
-    Expand-ZipArchive $scoopMainZipfile $scoopMainUnzipTempDir
-    Copy-Item "$scoopMainUnzipTempDir\Main-*\*" $SCOOP_MAIN_BUCKET_DIR -Recurse -Force
-
-    # Cleanup
-    Remove-Item $scoopUnzipTempDir -Recurse -Force
-    Remove-Item $scoopZipfile
-    Remove-Item $scoopMainUnzipTempDir -Recurse -Force
-    Remove-Item $scoopMainZipfile
 
+    if (Test-Command-Available('git')) {
+        if ($downloader.Proxy) {
+            #define env vars for git when behind a proxy
+            $Env:HTTP_PROXY = $downloader.Proxy.Address
```

Comment:
@chawyehsu Hi there! Sorry, it took a while until I found some time. My current implementation uses the given proxy to create env vars for git. What do you think about that?

Reply:
Hi @xxthunder , finally took me some time to review this change, env vars are fine. A thing we have to take care of is these env vars may be leaked to the current PS session, hence there should be some code to recover `$env:HTTP_PROXY` and `$env:HTTPS_PROXY`.

There is an [`Invoke-Git`](https://github.com/ScoopInstaller/Scoop/blob/0a39de86e2900f2e2bf3ddae1390a57b7d2c894f/lib/core.ps1#L157-L177) cmdlet in Scoop Core you may have a look about env vars recovery.

Reply:
@chawyehsu Finally found some time again. Thanks for pointing me to this Invoke-Git function. So I took over the env vars recovery. Tests are green: https://github.com/xxthunder/ScoopInstall/actions/runs/4368498689